int32 val
int32 SUCCESS=1
int32 FAILURE=99999
int32 INVALID=-1

int32 NOSTATE = -2
int32 NOCONTROLLER = -3
int32 NOTSTARTED = -4
int32 NOTSTOPPED = -5
